assets/setup-wizard/ready/use-sample-course.js

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+import { buildJobEndpointUrl } from '../../data-port/import/helpers/url';
+import { logEvent } from '../log-event';
+
+const useSampleCourse = () => {
+	const [ isInstalling, setIsInstalling ] = useState( false );
+	const [ error, setError ] = useState( null );
+	const [ jobId, setJobId ] = useState( null );
+	const [ pollingCount, setPollingCount ] = useState( 0 );
+
+	const pollingInterval = 2000;
+
+	const catchError = ( e ) => {
+		setIsInstalling( false );
+		setError( e.message );
+		setJobId( null );
+	};
+
+	// Logs polling.
+	useEffect( () => {
+		if ( ! jobId ) {
+			return;
+		}
+
+		const timer = setTimeout( () => {
+			apiFetch( {
+				path: buildJobEndpointUrl( jobId ),
```

Comment:
This could use the new `POST <jobId>/process` endpoint (which runs the job in the foreground), to speed things up. 
That also means that the request would be running for a few seconds, so we can even remove the polling interval timeout from the frontend, and start a new request as soon as one finishes.

Reply:
Good catch and well remembered! https://github.com/Automattic/sensei/pull/3507/commits/a1d3a7d0faff508a31724406f7868f45a0882f6c